src/main/java/com/github/manolo8/darkbot/core/manager/StatsManager.java

```diff
@@ -299,4 +335,47 @@ public String toString() {
                     "\nAverage=" + ONE_PLACE_FORMAT.format(getAverage());
         }
     }
+
+    public enum BootyKeyType {
+        GREEN(0x54),
+        BLUE(0x58),
+        RED(0x5c),
+        SILVER(0x60),
+        APOCALYPSE(0x64),
+        PROMETHEUS(0x68),
+        OBSIDIAN_MICROCHIP(0x6c),
+        BLACK_LIGHT_CODE(0x70),
+        BLACK_LIGHT_DECODER(0x74),
+        PROSPEROUS_FRAGMENT(0x78),
+        ASTRAL(0x7c),
+        ASTRAL_SUPREME(0x80),
+        EMPYRIAN(0x84),
+        LUCENT(0x88),
+        PERSEUS(0x8c);
+
+        private int offset;
+
+        private BootyKeyType(int offset) {
+            this.offset = offset;
```

Comment:
```suggestion
        private final int offset;
        private final StatsAPI.Key key;
        private StatImpl stat;

        private BootyKeyType(int offset, StatsAPI.Key statKey) {
            this.offset = offset;
```
If you add the stat key, and simply pass in `Stats.BootyKey.RED` etc in the constructor, you can more directly correlate the stats enum to this enum.

Then you can add a simple:

```java
public void track(StatsManager stats) {
  if (stat == null) stat = stats.get(StatKey.of(key));
  stat.track(stats.readDouble(offset));
}
```
in the enum, and call it for each of the enum constants

Reply:
Updated